### ISSUE TITLE
Clarify GITHUB_TOKEN rights

### ISSUE THREAD

**Issue by dbischof90** [mikepenz/release-changelog-builder-action#1223]
Hi! I'm setting this action up for a pipeline based on Gitea, not Github. For this I need to understand which rights you require from `GITHUB_TOKEN` so that I can replicate this on my side.

If you'd create a PAT, which rights would it need?

**Comment by mikepenz** [comment]
@dbischof90 This slightly depends on the specific configuration. However, the action only requires `read` rights as it does not create a release. It solely focuses on creating the release notes, and the usage for that is afterward open for the specific usecase. 


- `read` for [Pull requests](https://github.com/mikepenz/release-changelog-builder-action/blob/develop/pr-collector/src/pullRequests.ts#L124)
  - that should be covered by the `pull-requests` scope
- `read` for [Commits](https://github.com/mikepenz/release-changelog-builder-action/blob/develop/pr-collector/src/commits.ts#L54) 
  - that should be covered by the `contents` scope
- `read` for [Tags](https://github.com/mikepenz/release-changelog-builder-action/blob/develop/pr-collector/src/tags.ts#L32) (if not available the `from` and `to` refs have to be provided)
  - that should be covered by the `contents` scope
- `read` to [list reviews](https://github.com/mikepenz/release-changelog-builder-action/blob/develop/pr-collector/src/pullRequests.ts#L186)
  - that should be covered by the `contents` scope
 

It's worth noting that `pull requests` may not be required if the `commitMode` is enabled.


I don't see any other needs. Usually the above are covered by default Github Actions token scope.

**Comment by dbischof90** [comment]
Hm, that doesn't seem to be it... That's the current output I see in `act_runner`:

```
[...]
::group::📘 Reading input values
GITHUB_WORKSPACE = 'C:\Users\xxx\.cache\act\xxx\hostexecutor'
repositoryPath = 'C:\Users\xxx\.cache\act\xxx\hostexecutor'
Configuration file not provided.
ℹ️ No configuration provided. Using Defaults.
Resolved 'owner' as xxx
Resolved 'repo' as xxx
::endgroup::
::group::🔖 Resolve tags
::error::Bad credentials
[...]
``` 

I created a token with read rights to pretty much everything besides admin stuff:
![grafik](https://github.com/mikepenz/release-changelog-builder-action/assets/8844723/567fe7dd-cae9-41c1-a32a-ee7e5d534bcb)
Seems to be a bit more to that. Any guesses?

**Comment by dreamncn** [comment]
@dbischof90 

I apologize for replying under a closed issue.

This repository uses the `compareCommits` API of `octokit`. You can find the implementation here: [link](https://github.com/mikepenz/release-changelog-builder-action/pr-collector/src/commits.ts#L58C54-L58C68).

In `octokit`, it makes an HTTP request to `/repos/{owner}/{repo}/compare/{base}...{head}`. You can see the code for this request here: [link](https://github.com/octokit/plugin-rest-endpoint-methods.js/blob/da7daee52be3c96e3236d2a93e5fcabbe99ae95d/src/generated/parameters-and-response-types.ts#L4854C26-L4854C72).

However, Gitea does not have an API named `/repos/{owner}/{repo}/compare/`. It may be necessary to develop a new library to replace `octokit`.

**Comment by mikepenz** [comment]
Thank you for adding additional context. If somebody knows how to work around that limitation I could look into that, but as mentioned before I don't have experience with gitea so I wouldn't be able to test it.

**Comment by dreamncn** [comment]
> Thank you for adding additional context. If somebody knows how to work around that limitation I could look into that, but as mentioned before I don't have experience with gitea so I wouldn't be able to test it.

I forked your repository and am extracting all the parts that involve octokit calls. This way, I can extend the Action to support more types of action CI/CD systems. Wait for my good news!

**Comment by dbischof90** [comment]
I did not expect that - great news! :)

**Comment by mikepenz** [comment]
@dreamncn thank you so much. Really appreciate you looking into this!

**Comment by dreamncn** [comment]
<img width="1521" alt="image" src="https://github.com/mikepenz/release-changelog-builder-action/assets/37787014/44df6ca7-de53-4dc2-897f-f895fc3f8e6e">
It seems to run fine on Gitea, I'll refine the test cases tomorrow before submitting a PR.

**Comment by mikepenz** [comment]
Wow that was fast. Impressive!

**Comment by dreamncn** [comment]
@mikepenz @dbischof90 I have successfully completed it. For details, see PR #1273

**Comment by mikepenz** [comment]
The PR was merged and is released as part of v4.1.0 @dbischof90 

Thanks for the awesome contribution @dreamncn
